exist-core/src/main/java/org/exist/client/InteractiveClient.java

```diff
@@ -377,39 +377,38 @@ protected void getResources() throws XMLDBException {
             return;
         }
         setProperties();
-        final UserManagementService mgtService = (UserManagementService) current
-                .getService("UserManagementService", "1.0");
-        final String childCollections[] = current.listChildCollections();
-        final String childResources[] = current.listResources();
+        final UserManagementService mgtService = current.getService(UserManagementService.class);
+        final List<String> childCollections = current.listChildCollections();
+        final List<String> childResources = current.listResources();
 
-        resources = new String[childCollections.length + childResources.length];
+        resources = new String[childCollections.size() + childResources.size()];
         //Collection child;
         Permission perm;
 
         final List<ResourceDescriptor> tableData = new ArrayList<>(resources.length); // A list of ResourceDescriptor for the GUI
 
         int i = 0;
-        for (; i < childCollections.length; i++) {
+        for (; i < childCollections.size(); i++) {
```

Comment:
looks like a better orlop can babe used, not using the index?

Reply:
I was too tiered to tackle this in the first pass  😊